client.go

```diff
@@ -48,6 +48,14 @@ func SelectIPTraffic(t IPType) ClientOption {
 	}
 }
 
+// SelectIfaces selects the interfaces to query for mDNS records
+func SelectIfaces(ifaces []net.Interface) ClientOption {
+	return func(o *clientOpts) {
+		o.ifaces = make([]net.Interface, len(ifaces), (cap(ifaces) + 1))
```

Comment:
Just make: `o.ifaces = ifaces`
It is copied implicitly. No need for explicit copy.